### PR TITLE
@types/mysql-import [update]: Update to version 5 definition.

### PR DESCRIPTION
**Pull request by doubutsunokarada**
Please fill in this template.

- [ X ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ X ] Test the change in your own code. (Compile and run.)
- [ X ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ X ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ X ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ X ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ X ] Provide a URL to documentation or source code which provides context for the suggested changes: [mysql-import](https://www.npmjs.com/package/mysql-import)
- [ X ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/mysql-import/index.d.ts
1. types/mysql-import/mysql-import-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 208206a4cec9ce4bd73f6d2606d4333d49d5ff99
</p>

File: types/mysql-import/index.d.ts [hunk 1]
```diff
@@ -1,35 +1,13 @@
...
 
-export function config(settings: Settings): Importer;
+    getImported(): Array<{file: string, size: number}>;
+    setEncoding(encoding: 'utf8' | 'ucs2' | 'utf16le' | 'latin1' | 'ascii' | 'base64' | 'hex'): void;
+    use(database: string): Promise<void>;
+    onProgress(cb: (...args: any[]) => any): void;
+    onDumpComplete(cb: (...args: any[]) => any): void;
+    import(input: string | string[]): Promise<void>;
+    disconnect(graceful: boolean): Promise<void>;
+}
```
Comment (bengry) [inline types/mysql-import/index.d.ts:13]:
```suggestion
    onProgress(callback: (object: {
      total_files: number;
		  file_no: number;
		  bytes_processed: number;
		  total_bytes: number;
		  file_path: string;
	  }) => void): void;
    onDumpComplete(callback: (object: {
      total_files: number;
		  file_no: number;
		  		  file_path: string;
		  		  error: Error | null;
}) => void): void;
    import(...input: ReadonlyArray<string | readonly string[]>): Promise<void>;
    disconnect(graceful?: boolean): Promise<void>;
}
```

**Review by bengry — CHANGES_REQUESTED** [review]

File: types/mysql-import/index.d.ts (hunk 1 above)
Comment (doubutsunokarada) [reply types/mysql-import/index.d.ts:13]: thanks!

File: types/mysql-import/index.d.ts [hunk 2]
```diff
@@ -1,35 +1,13 @@
...
+import { ConnectionOptions } from 'mysql2';
 
-export interface Importer {
-    /**
-     * Import an .sql file to the database.
-     */
-    import(filename: string): Promise<void>;
-}
+export default class Importer {
+    constructor(settings: Pick<ConnectionOptions, 'user' | 'password' | 'database' | 'host'>);
```
[the remark this answers is not in the document]
Comment (doubutsunokarada) [reply types/mysql-import/index.d.ts:4]: > What about port, ssl? (https://github.com/Pamblam/mysql-import#new-importerhost-user-password-database-port-ssl).

Sorry, I missed that.

> Why is the Pick needed?

`ConnectionOptions` has other properties that are not needed. Should I remove it?

**Review by bengry — APPROVED** [review]

**Comment by doubutsunokarada** [comment]
Ready to merge

**Comment by doubutsunokarada** [comment]
@bengry Thanks for reviewing!

**Comment by bengry** [comment]
> @bengry Thanks for reviewing!

Thanks for your work, and for sharing the types with the community 🥇

**Pull request merged**